### ISSUE TITLE
Model evaluation exploration UI has a regression from 5.5.1

### ISSUE THREAD

**Issue by srikris** [apple/turicreate#1932]
```
/Users/srikris/.virtualenvs/turicreate-5.6rc/lib/python2.7/site-packages/turicreate/toolkits/image_classifier/_evaluation.pyc in explore(self)
     49
     50   def explore(self):
---> 51     _thread.start_new_thread(_start_process, (self._get_eval_json()+"\n", self.data["test_data"], self, ))
     52
     53

/Users/srikris/.virtualenvs/turicreate-5.6rc/lib/python2.7/site-packages/turicreate/toolkits/image_classifier/_evaluation.pyc in _get_eval_json(self)
     46     evaluation_dictionary["incorrects_size"] = evaluation_dictionary["table_spec"]["size"] - evaluation_dictionary["corrects_size"]
     47
---> 48     return str(_json.dumps({ "evaluation_spec": evaluation_dictionary }, allow_nan = False))
     49
     50   def explore(self):

/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.pyc in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, encoding, default, sort_keys, **kw)
    248         check_circular=check_circular, allow_nan=allow_nan, indent=indent,
    249         separators=separators, encoding=encoding, default=default,
--> 250         sort_keys=sort_keys, **kw).encode(obj)
    251
    252

/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.pyc in encode(self, o)
    205         # exceptions aren't as detailed.  The list call should be roughly
    206         # equivalent to the PySequence_Fast that ''.join() would do.
--> 207         chunks = self.iterencode(o, _one_shot=True)
    208         if not isinstance(chunks, (list, tuple)):
    209             chunks = list(chunks)

/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.pyc in iterencode(self, o, _one_shot)
    268                 self.key_separator, self.item_separator, self.sort_keys,
    269                 self.skipkeys, _one_shot)
--> 270         return _iterencode(o, 0)
    271
    272 def _make_iterencode(markers, _default, _encoder, _indent, _floatstr,

/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.pyc in default(self, o)
    182
    183         """
--> 184         raise TypeError(repr(o) + " is not JSON serializable")
    185
    186     def encode(self, o):

TypeError: Columns:
	target_label	str
	predicted_label	str
	count	int

Rows: 41

Data:
+--------------+-----------------+-------+
| target_label | predicted_label | count |
+--------------+-----------------+-------+
|   ellipse    |     ellipse     |   9   |
|  chatBubble  |     freeform    |   8   |
|    heart     |      arrow      |   1   |
|   freeform   |       line      |   8   |
|    arrow     |     freeform    |   7   |
|   freeform   |     triangle    |   1   |
|    cloud     |     freeform    |   2   |
|     star     |     freeform    |   1   |
|     line     |     ellipse     |   2   |
|   freeform   |       star      |   4   |
```

**Comment by znation** [comment]
Repro steps?

**Comment by znation** [comment]
Not a regression; this is fixed in #1903 and works in current master.
